### ISSUE TITLE
Feature request: Implement ISteamUser::GetAuthTicketForWebApi

### ISSUE THREAD

**Issue by LZQCN** [ceifa/steamworks.js#118]
Steamworks.js is a great JavaScript library. However, it lacks implementation of the ISteamUser::GetAuthTicketForWebApi method, which is crucial for my game's server-side interface to only allow access to verified Steam users.

Thank you very much.

**Comment by flyinghail** [comment]
The same request.

I looked at #52 and tested it and realized that this would not work. 

I looked at the source code `getSessionTicket()` uses `client.user().authentication_session_ticket()`, calls `SteamAPI_ISteamUser_GetAuthSessionTicket`.

Server authentication required `client.user().authentication_session_ticket_for_webapi()`

**Comment by flyinghail** [comment]
I tried to modify myself to the latest `steamworks-rs` and wrap `GetAuthTicketForWebApi`
It compiles fine, but the ticket doesn't pass the [ISteamUserAuth/AuthenticateUserTicket](https://partner.steamgames.com/doc/webapi/ISteamUserAuth#AuthenticateUserTicket) validation.

I'm not very familiar with rust, so I'm not sure what I should change next.

The source code likes:

```rust
    #[napi]
    pub async fn get_session_ticket_for_webapi(identity: String, timeout_seconds: Option<u32>) -> Result<Ticket, Error> {
        let client = crate::client::get_client();
        let (tx, rx) = oneshot::channel();
        let mut tx = Some(tx);

        let ticket_handle = client.user().authentication_session_ticket_for_webapi(&identity);

        let callback =
            client.register_callback(move |ticket_for_webapi_response: TicketForWebApiResponse| {
                if ticket_for_webapi_response.ticket_handle == ticket_handle {
                    let mut ticket = ticket_for_webapi_response.ticket;
                    ticket.truncate(ticket_for_webapi_response.ticket_len as usize);

                    if let Some(tx) = tx.take() {
                        tx.send(match ticket_for_webapi_response.result {
                            Ok(()) => Ok(ticket),
                            Err(e) => Err(Error::from_reason(e.to_string())),
                        })
                            .unwrap();
                    }
                }
            });

        let timeout_seconds = u64::from(timeout_seconds.unwrap_or(10));
        let result =
            tokio::time::timeout(std::time::Duration::from_secs(timeout_seconds), rx).await;

        drop(callback);

        match result {
            Ok(result) => match result {
                Ok(Ok(data)) => Ok(Ticket {
                    handle: ticket_handle,
                    data,
                }),
                Ok(Err(e)) => {
                    client.user().cancel_authentication_ticket(ticket_handle);
                    Err(e)
                }
                Err(e) => {
                    client.user().cancel_authentication_ticket(ticket_handle);
                    Err(Error::from_reason(e.to_string()))
                }
            },
            Err(_) => {
                client.user().cancel_authentication_ticket(ticket_handle);
                Err(Error::from_reason(
                    "Steam didn't validated the ticket in time.",
                ))
            }
        }
    }
```

**Comment by guohaolay** [comment]
+1 to this feature request.
This package is such a hidden gem, super simple to use and works well. Could we bother the maintainers migrate to using the `authentication_session_ticket_for_webapi`? Steam have requested all their clients to migrate on May 2nd, 2023.

**Comment by LZQCN** [comment]
I have learned Rust and plan to fork this repository within this month. I will implement APIs such as `workshop_items::query_all`, `auth::get_auth_ticket_for_web_api`, `workshop::update_item_cb`, etc.

**Comment by LZQCN** [comment]
I have created two forks: `ai-zen/steamworks-rs` and `ai-zen/steamworks.js`.
The upstream repository for `steamworks.js` is `steamworks-rs`, but it has compilation errors.
I fixed the compilation errors in `ai-zen/steamworks-rs`, but the PR has not been approved yet.
To ensure that the new changes can be immediately used in production, I changed the upstream repository for `ai-zen/steamworks.js` to `ai-zen/steamworks-rs`.
I also released a separate npm package called `@ai-zen/steamworks.js`.
After the upstream PR is completed, I will submit a PR to merge `ai-zen/steamworks.js` into `steamworks.js`.
